ludwig/hyperopt/run.py

```diff
@@ -441,11 +441,11 @@ def get_features_eligible_for_shared_params(
             raise ValueError("Ludwig expects feature types to be defined for each feature within the config.")
         if config_feature_type == INPUT_FEATURES:
             default_encoder = get_default_encoder_or_decoder(feature, INPUT_FEATURES)
```

Comment:
@connor-mccorm I have a feeling this function might need modifications since it infers type directly from `feature.get('type')` instead of `feature.get('encoder').get('type')` - might just want to double check